src/components/code-editor/code-editor.scss

```diff
@@ -9,135 +9,134 @@
+    --code-editor-selection-lowlight-color: var(--contrast-600);
+
+    .cm-attribute,
+    .cm-property,
+    .cm-keyword {
+        color: rgb(var(--color-violet-light));
+    }
+    .cm-operator {
+        color: rgb(var(--color-sky-dark));
+    }
+    .cm-variable {
+        color: rgb(var(--color-pink-light));
+    }
+    .cm-variable-2 {
+        color: rgb(var(--contrast-1100));
+    }
+    .cm-variable-3,
+    .cm-type {
+        color: rgb(var(--color-amber-darker));
+    }
+    .cm-builtin,
+    .cm-meta {
+        color: rgb(var(--color-amber-default));
+    }
+    .cm-atom {
+        color: rgb(var(--color-coral-default));
+    }
+    .cm-tag,
+    .cm-number {
+        color: rgb(var(--color-magenta-default));
+    }
+    .cm-def {
+        color: rgb(var(--color-sky-default));
+    }
+    .cm-string {
+        color: rgb(var(--color-green-dark));
+    }
+    .cm-string-2 {
+        color: rgb(var(--color-pink-default));
+    }
+    .cm-qualifier {
+        color: rgb(var(--color-grass-default));
+    }
+    .cm-error {
+        color: rgb(var(--color-white));
+        background-color: rgb(var(--color-red-default));
+    }
+    .cm-comment {
+        color: rgba(var(--code-editor-neutral-text-color), 0.7);
+    }
+}
+
+@mixin dark-mode-styles {
+    --code-editor-neutral-text-color: var(--contrast-1600);
+    // --code-editor-background-color: { same as light };
+    // --code-editor-gutter-background-color: { same as light };
+    --code-editor-line-number-text-color: var(--contrast-800);
+    --code-editor-cursor-color: var(--color-amber-default);
+
+    --code-editor-selection-highlight-color: var(--color-cyan-light);
+    --code-editor-selection-lowlight-color: var(--contrast-100);
+
+    .cm-attribute,
+    .cm-property,
+    .cm-keyword {
+        color: rgb(var(--color-purple-lighter));
+    }
+    .cm-operator {
+        color: rgb(var(--color-sky-lighter));
+    }
+    // .cm-variable { same as light }
+    // .cm-variable-2 { same as light }
+    .cm-variable-3,
+    .cm-type {
+        color: rgb(var(--color-yellow-light));
+    }
+    // .cm-builtin,
+    // .cm-meta { same as light }
+    .cm-atom {
+        color: rgb(var(--color-coral-light));
+    }
+    .cm-tag,
+    .cm-number {
+        color: rgb(var(--color-magenta-light));
+    }
+    .cm-def {
+        color: rgb(var(--color-sky-light));
+    }
+    .cm-string {
+        color: rgb(var(--color-lime-light));
+    }
+    .cm-string-2 {
+        color: rgb(var(--color-pink-lighter));
+    }
+    .cm-qualifier {
+        color: rgb(var(--color-grass-light));
+    }
+    // .cm-error { same as light }
+    .cm-comment {
+        color: rgba(var(--code-editor-neutral-text-color), 0.4);
+    }
+}
+
 :host {
     display: flex;
     font-size: var(--code-editor-font-size, 0.875rem); // 14px
+    @include light-mode-styles;
+}
+
+@include mixins.in(dark-mode) {
+    :host {
+        @include dark-mode-styles;
+    }
 }
```

Comment:
This one of course doesn't work, and I have no idea how to solve it.
The problem is we want certain styles for the dark-mode of the code editor.
We don't want to simply switch the colors to their dark-mode variants.
However, our `in(dark-mode)` mixin doesn't work, since it's targeting the `:root` and we can't do it from within the component.
Maybe we should add `data-theme='force-light'` and `data-theme='force-dark'` to the `<Host />` of limel-code-editor (instead of those classes of `.is-dark-mode` and `.is-light-mode` which we currently add); and then use some CSS like below to style the dark mode 🤔 I don't know. What do you say?

```scss
@media (prefers-color-scheme: dark) {
            :host(:not([data-theme='force-light'])) & {
                @content;
            }
        }
        :host([data-theme='force-dark']) & {
            @content;
        }
```

Reply:
I'll try to remember to look into this.

Reply:
Did this get solved? The code editor seemed to work just fine in dark and light mode? Or did I miss something? I primarily checked the light mode, since that's where there could be breaking changes…

Reply:
oh crap. no

Reply:
I have no idea how to solve it. 😢